### PR TITLE
"iceberg" to "healthbot" changes and PR fixes

### PR DESCRIPTION
**Pull request by gvarao**

### CODE REVIEW

**Comment by ayushgithub** [comment]
jenkins please retry a build

**Comment by gvarao** [comment]
jenkins please retry a build

**Comment by gvarao** [comment]
jenkins please retry a build

File: community_supplied/Chassis/chassis-env-info.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
```
Comment (gaddevijay) [inline community_supplied/Chassis/chassis-env-info.yml]: Please do not dump all the changes in a single Pull request. Request you to segregate them logically and submit appropriate requests

Comment (gvarao) [reply community_supplied/Chassis/chassis-env-info.yml]: Done

File: juniper_official/Linecard/check-pfe-discards.rule [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+                type string;
+                description "PFE number";
+            }
+            trigger data-error-discard {
+                synopsis "PFE data error discards KPI";
+                description "Sets health based on the change in PFE data error discard count";
+                frequency 60s;
+                term is-data-error-discard {
+                    when {
+                        increasing-at-least-by-value "$data-error-discard" {
```
Comment (ConstrainedRandomVar) [inline juniper_official/Linecard/check-pfe-discards.rule]: Shouldn't we be using a time-range here to ensure that we deal with jitter/delays in iAgent and writing to Influx?

File: juniper_official/Chassis/chassis-alarms.rule [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            trigger chassis-alarms {
+                synopsis "Chassis alarms KPI";
+                description "Sets health based on chassis alarm type";
+                frequency 60s;
+                /*
+                * Sets color to red and sends out an anomaly notification when
+                * major chassis alarms detected
+                * 
+                */
+                term is-major-alarm-exists {
```
Comment (ConstrainedRandomVar) [inline juniper_official/Chassis/chassis-alarms.rule]: Please change this to does-major-alarm-exist.

File: juniper_official/Linecard/check-pfe-discards.rule [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+                    }
+                }
+            }
+            trigger pfe-bad-route-discard {
+                synopsis "PFE bad route discards KPI";
+                description "Sets health based on the change in PFE bad route discard count";
+                frequency 60s;
+                term is-bad-route-discard {
+                    when {
+                        increasing-at-least-by-value "$bad-route-discard";
```
Comment (ConstrainedRandomVar) [inline juniper_official/Linecard/check-pfe-discards.rule]: Should we be using a time-range here?

File: juniper_official/Linecard/check-pfe-discards.rule (hunk 2 above)
Comment (ConstrainedRandomVar) [inline juniper_official/Linecard/check-pfe-discards.rule]: Would it make sense to declare all the iAgent YAML implicit fields in the rule explicitly to ensure that these can be plotted as part of the field table?

File: juniper_official/Linecard/check-pfe-discards.rule (hunk 4 above)
Comment (ConstrainedRandomVar) [inline juniper_official/Linecard/check-pfe-discards.rule]: Can we introduce $bad-route-discard here as a field explicitly?

File: juniper_official/Protocols/Isis/isis-statistics.rule [hunk 5]
```diff
@@ -72,6 +75,7 @@
                 sensor isis-sensor {
                     where "/network-instances/network-instance/protocols/protocol/isis/interfaces/interface/@interface-id =~ /{{interface-name}}/";
                     path "/network-instances/network-instance/protocols/protocol/isis/interfaces/interface/@interface-id";
+                    zero-suppression;
```
Comment (ConstrainedRandomVar) [inline juniper_official/Protocols/Isis/isis-statistics.rule]: Why are we zero-suppressing something of type string? Can you please double check this?

File: juniper_official/Protocols/Lldp/lldp-session-statistics.rule [hunk 6]
```diff
@@ -118,20 +121,23 @@ healthbot {
                 sensor lldp-sensor {
                     where "/lldp/interfaces/interface/@name =~ /{{if-name}}/";
                     path "/lldp/interfaces/interface/@name";
+                    zero-suppression;
```
Comment (ConstrainedRandomVar) [inline juniper_official/Protocols/Lldp/lldp-session-statistics.rule]: Same here. Should we be zero-suppressing type string?

File: juniper_official/Security/security-kpis-playbook.playbook [hunk 7]
```diff
@@ -0,0 +1,7 @@
+healthbot {
+    playbook security-kpis-playbook {
+        rules [ security/check-veriexec-status security/check-veriexec-status-iagent ];
+        description "This playbook all the rules keeping a check on security parameters from the topic security" ;
```
Comment (ConstrainedRandomVar) [inline juniper_official/Security/security-kpis-playbook.playbook]: Please rephrase description.

File: juniper_official/System/system-cpu.rule [hunk 8]
```diff
@@ -69,6 +69,7 @@ healthbot {
                 sensor components {
                     where "/components/component/properties/property/@name == 'cpu-utilization-idle'";
                     path /components/component/properties/property/state/value;
+                    zero-suppression;
```
Comment (ConstrainedRandomVar) [inline juniper_official/System/system-cpu.rule]: If we want to zero-suppress cpu-utilization-idle, can we first declare it as type int?

File: juniper_official/Linecard/check-pfe-discards.rule (hunk 2 above)
Comment (gvarao) [reply juniper_official/Linecard/check-pfe-discards.rule]: It has been handled in iAgent hence we can plot graphs for all fields below;    fabric-discard: {fabric-discard: int}
    bits-to-test-discard: {bits-to-test-discard: int}
    data-error-discard: {data-error-discard: int}
    tcp-header-error-discard: {tcp-header-error-discard: int}
    stack-underflow-discard: {stack-underflow-discard: int}
    stack-overflow-discard: {stack-overflow-discard: int}
    bad-route-discard: {bad-route-discard: int}
    nexthop-discard: {nexthop-discard: int}
    invalid-iif-discard: {invalid-iif-discard: int}
    info-cell-discard: {info-cell-discard: int}

Comment (gvarao) [reply juniper_official/Linecard/check-pfe-discards.rule]: Discards should be evaluated every time(sensor frequency) we receives data as these are critical hence time-range is not required.

File: juniper_official/Linecard/check-pfe-discards.rule (hunk 4 above)
Comment (gvarao) [reply juniper_official/Linecard/check-pfe-discards.rule]: Discards should be evaluated every time(sensor frequency) we receives data as these are critical hence time-range is not required.

Comment (gvarao) [reply juniper_official/Linecard/check-pfe-discards.rule]: $bad-route-discard explicitly defined as integer in yml as shown below;

bad-route-discard: {bad-route-discard: int}

File: juniper_official/Protocols/Isis/isis-statistics.rule (hunk 5 above)
Comment (gvarao) [reply juniper_official/Protocols/Isis/isis-statistics.rule]: Corrected.

File: juniper_official/Protocols/Lldp/lldp-session-statistics.rule (hunk 6 above)
Comment (gvarao) [reply juniper_official/Protocols/Lldp/lldp-session-statistics.rule]: Corrected

File: juniper_official/Security/security-kpis-playbook.playbook (hunk 7 above)
Comment (gvarao) [reply juniper_official/Security/security-kpis-playbook.playbook]: Modified and committed.

File: juniper_official/System/system-cpu.rule (hunk 8 above)
Comment (gvarao) [reply juniper_official/System/system-cpu.rule]: Corrected

File: juniper_official/Chassis/chassis-alarms.rule (hunk 3 above)
Comment (gvarao) [reply juniper_official/Chassis/chassis-alarms.rule]: Corrected

File: juniper_official/Linecard/check-pfe-discards.rule (hunk 2 above)
Comment (ConstrainedRandomVar) [reply juniper_official/Linecard/check-pfe-discards.rule]: Hi Venkat,
 The issue is that both sensor and trigger frequency are set to 60s. That means that for the lookback window for checking for data is 120s - which doesn't necessarily guarantee the presence of two data points. Ideally this should be 3x.

Comment (gvarao) [reply juniper_official/Linecard/check-pfe-discards.rule]: As we discussed increased trigger frequency to 75s

**Pull request merged**